tests/b2share_unit_tests/records/test_records_api.py

```diff
@@ -202,3 +202,40 @@ def test_record_delete_version(app, test_records, test_users):
 
         # TODO: test in ES that the previous version is reindexed
         # and can be found
+
+
+def test_record_publish_adds_no_handles_for_external_files(app,
+                            records_data_with_external_pids,
+                            test_records_data):
+    """Test that no handle PIDs are created for external files."""
+    for metadata in test_records_data:
+        with app.app_context():
+            app.config.update({'FAKE_EPIC_PID': True})
+
+            external_pids = records_data_with_external_pids['external_pids']
+            external_keys = {x['key'] for x in external_pids}
+            data = deepcopy(metadata)
+            data['external_pids'] = deepcopy(external_pids)
+
+            record_uuid = uuid.uuid4()
+            b2share_deposit_uuid_minter(record_uuid, data=data)
+
+            deposit = Deposit.create(data, id_=record_uuid)
+            ObjectVersion.create(deposit.files.bucket, 'real_file_1.txt',
+                             stream=BytesIO(b'mycontent'))
+            ObjectVersion.create(deposit.files.bucket, 'real_file_2.txt',
+                             stream=BytesIO(b'mycontent'))
+            deposit.submit()
+            deposit.publish()
+            deposit.commit()
+
+            _, record = deposit.fetch_published()
+
+            # external files don't get a handle PID, they already have one
+            # which is stored in record['_deposit']['external_pids']
+            for f in record.files:
+                if f['key'] in external_keys:
+                    assert f.get('ePIC_PID') is None
```

Comment:
Won't the UI be missing the ePIC_PID in `_files`? We could also just copy the PID from external_pids.

Reply:
That is true. Right now we don't show any pids for external files. The tests would have to be a bit different then.

Reply:
We can just change the PR by copying the PID from external_pids, which simplifies the work of 3rd party scripts. They will always be able to retrieve the ePIC_PID in `_files` that way.